versioned/mongodb/src/test/java/com/dremio/nessie/versioned/impl/SampleEntities.java

```diff
@@ -0,0 +1,225 @@
+    final List<Entity> treeList = new ArrayList<>(L1.SIZE);
+    for (int i = 0; i < L1.SIZE; i++) {
+      treeList.add(createIdEntity());
+    }
+
+    // Two commits, one with a DELTA and one without.
+    final List<Entity> commitList = new ArrayList<>(2);
+    commitList.add(Entity.ofMap(ImmutableMap.of(
+        "id", createIdEntity(),
+        "commit", createIdEntity(),
+        "parent", createIdEntity()
+    )));
+    commitList.add(Entity.ofMap(ImmutableMap.of(
+        "id", createIdEntity(),
+        "commit", createIdEntity(),
+        "deltas", Entity.ofList(Entity.ofMap(ImmutableMap.of(
+            "position", Entity.ofNumber(1),
+            "old", createIdEntity(),
+            "new", createIdEntity()
+        ))),
+        "keys", Entity.ofList(Entity.ofMap(ImmutableMap.of("a",
+            Entity.ofList(createStringEntity(8), createStringEntity(8)))))
+    )));
+
+    final Map<String, Entity> attributeMap = new HashMap<>();
+    attributeMap.put(InternalRef.TYPE, Entity.ofString("b"));
+    final String name = createString(10);
+    attributeMap.put("name", Entity.ofString(name));
+    attributeMap.put("id", Id.build(name).toEntity());
+    attributeMap.put("tree", Entity.ofList(treeList));
+    attributeMap.put("metadata", createIdEntity());
+    attributeMap.put("commits", Entity.ofList(commitList));
+
+    return InternalRef.SCHEMA.mapToItem(attributeMap);
+  }
+
+  /**
+   * Create a Sample Tag (InternalRef) entity.
+   * @return sample Tag (InternalRef) entity.
+   */
+  public static InternalRef createTag() {
+    final Map<String, Entity> attributeMap = new HashMap<>();
+    attributeMap.put(InternalRef.TYPE, Entity.ofString("t"));
+    attributeMap.put("id", createIdEntity());
+    attributeMap.put("name", Entity.ofString("tagName"));
+    attributeMap.put("commit", createIdEntity());
+
+    return InternalRef.SCHEMA.mapToItem(attributeMap);
+  }
+
+  /**
+   * Create a Sample CommitMetadata entity.
+   * @return sample CommitMetadata entity.
+   */
+  public static InternalCommitMetadata createCommitMetadata() {
+    final Map<String, Entity> attributeMap = new HashMap<>();
+    attributeMap.put("id", createIdEntity());
+    attributeMap.put("value", Entity.ofBinary(createBinary(6)));
+
+    return InternalCommitMetadata.SCHEMA.mapToItem(attributeMap);
+  }
+
+  /**
+   * Create a Sample Value entity.
+   * @return sample Value entity.
+   */
+  public static InternalValue createValue() {
+    final Map<String, Entity> attributeMap = new HashMap<>();
+    attributeMap.put("id", createIdEntity());
+    attributeMap.put("value", Entity.ofBinary(createBinary(7)));
+
+    return InternalValue.SCHEMA.mapToItem(attributeMap);
+  }
+
+  /**
+   * Create an array of random bytes.
+   * @param numBytes the size of the array.
+   * @return the array of random bytes.
+   */
+  public static byte[] createBinary(int numBytes) {
+    final byte[] buffer = new byte[numBytes];
+    new Random().nextBytes(buffer);
```

Comment:
Can you update this so we make sure each test always runs with the same random seeds?

Reply:
The best way is probably the public methods receive an input of a random or seed.